### ISSUE TITLE
Train using loss function that accepts whole images per batch

### ISSUE THREAD

**Issue by Mason-McGough** [nerfstudio-project/nerfstudio#1419]
I would like to experiment with using a loss function that takes a whole image in as input. However, the training takes bundles of rays in as forward passes rather than whole images. Is there a way to train on single images at a time?

So far I have experimented with the following method:
1. Start training using `--pipeline.datamanager.train-num-rays-per-batch 16384` and `--pipeline.datamanager.train-num-images-to-sample-from 1` for images of size 128x128.
2. Add a breakpoint to the `get_loss_dict` method of `NerfactoModel`
3. Reshape the rays to an image using `outputs['rgb'].view(128, 128, 3)`

The image looks like this:
![scrambled-rays](https://user-images.githubusercontent.com/14284110/218596644-a798670e-adc1-445b-b401-b78703714607.png)

Issue is the results are garbled as if the rays are not in order. Possibly they are randomly sampled? Can I train with batches of the rays from a single image in order? I realize this won't scale well for larger images, but for now 128x128 is fine for my use case.

**Comment by tancik** [comment]
It's not in the main repo, but here is some code to sample patches - https://github.com/nerfstudio-project/nerfstudio/blob/e36bed250c90cfaac18e77b04cf340776b0bf9f7/nerfstudio/data/pixel_samplers.py#L248

**Comment by duxiaodan** [comment]
Hi @tancik I have a related question: I also want to batch all rays in a single image as input rather than random rays from all images. So I modify the datamanager so that now it outputs a ray bundle for a single image in row-first sequence. However, during testing, I found that by merely doing this, the GPU memory consumption increases by 75%. The batch size is the same, the numbers of True in occupancy_grid._binary are also pretty similar. Could you please advice me on what might be the problem? Thank you
